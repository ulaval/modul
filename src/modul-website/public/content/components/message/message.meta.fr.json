{
    "name": "Message",
    "url": "message",
    "category": "content",
    "type": "components",
    "beta": false,
    "visible": true,
    "overview": "components/message/message.overview.fr.md",
    "preview": "components/message/message.preview.fr.md",
    "components": [
        "m-message"
    ]
}
