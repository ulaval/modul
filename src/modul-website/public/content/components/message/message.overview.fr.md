Les *messages* sont utilisés pour donner de l'information pertinente aux utilisateurs.

## Caractéristiques

### Format du message

#### Régulier
Ce format est à privilégier lorsque le message concerne une page ou une section de page. La largeur du *message* varie en fonction de la largeur maximale du contenant dans lequel le message est affiché.

<modul-demo>

```html
<m-message>Ad labore ad ex do minim qui eu aliquip aute non. Tempor Lorem ipsum ad id esse minim. Pariatur cupidatat qui qui reprehenderit voluptate aute officia consectetur cillum sint do. Consequat ipsum ex est culpa sint velit nostrud nisi minim tempor nostrud tempor adipisicing do.</m-message>
```

</modul-demo>

#### Allégé
Ce format est à privilégier lorsque le message concerne un élément spécifique de l'interface. Il est possible d'ajouter un lien en fin de message, permettant de faire afficher le l'information supplémentaire dans une infobulle.

<modul-demo>

```html
<m-message skin="light">Ad labore ad ex do minim qui eu aliquip aute non. Tempor Lorem ipsum ad id esse minim. Pariatur cupidatat qui qui reprehenderit voluptate aute officia consectetur cillum sint do. Consequat ipsum ex est culpa sint velit nostrud nisi minim tempor nostrud tempor adipisicing do.</m-message>
```

</modul-demo>

### Type de message
Différents types de messages sont disponibles en fonction du type d'information à transmettre. Le type de message affiché se distingue par sa couleur et son icône.

#### Confirmation
Ce type de *message* permet d'indiquer à l'utilisateur qu'il a complété une demande, franchi une étape ou encore terminé un processus avec succès.

<modul-demo>

```html
<m-message state="confirmation">Ad labore ad ex do minim qui eu aliquip aute non. Tempor Lorem ipsum ad id esse minim. Pariatur cupidatat qui qui reprehenderit voluptate aute officia consectetur cillum sint do. Consequat ipsum ex est culpa sint velit nostrud nisi minim tempor nostrud tempor adipisicing do.</m-message>
```

</modul-demo>

#### Information
Ce type de *message* est utilisé lorsque l'on désire afficher de l'aide, une astuce ou tout simplement de l'information pertinente sur laquelle on désire mettre l'accent.

<modul-demo>

```html
<m-message state="information">Ad labore ad ex do minim qui eu aliquip aute non. Tempor Lorem ipsum ad id esse minim. Pariatur cupidatat qui qui reprehenderit voluptate aute officia consectetur cillum sint do. Consequat ipsum ex est culpa sint velit nostrud nisi minim tempor nostrud tempor adipisicing do.</m-message>
```

</modul-demo>

#### Avertissement
Ce type de *message* permet d'attirer l'attention de l'utilisateur et de l'aviser d'une situation inhabituelle.

<modul-demo>

```html
<m-message state="warning">Ad labore ad ex do minim qui eu aliquip aute non. Tempor Lorem ipsum ad id esse minim. Pariatur cupidatat qui qui reprehenderit voluptate aute officia consectetur cillum sint do. Consequat ipsum ex est culpa sint velit nostrud nisi minim tempor nostrud tempor adipisicing do.</m-message>
```

</modul-demo>

#### Erreur
Ce type de *message* est utilisé pour afficher des messages d'erreurs.

<modul-demo>

```html
<m-message state="error">Ad labore ad ex do minim qui eu aliquip aute non. Tempor Lorem ipsum ad id esse minim. Pariatur cupidatat qui qui reprehenderit voluptate aute officia consectetur cillum sint do. Consequat ipsum ex est culpa sint velit nostrud nisi minim tempor nostrud tempor adipisicing do.</m-message>
```

</modul-demo>

### Bouton de fermeture
Pour tous les types de messages au format régulier, il est possible d'ajouter une option pour le faire disparaître. De plus, il est possible d'effectuer certaines actions lors de la fermeture du message. L'utilisation de l'option de fermeture est à la discrétion de l'analyste.

<modul-demo>

```html
<m-message :close-button="true">Ad labore ad ex do minim qui eu aliquip aute non. Tempor Lorem ipsum ad id esse minim. Pariatur cupidatat qui qui reprehenderit voluptate aute officia consectetur cillum sint do. Consequat ipsum ex est culpa sint velit nostrud nisi minim tempor nostrud tempor adipisicing do.</m-message>
```

</modul-demo>
