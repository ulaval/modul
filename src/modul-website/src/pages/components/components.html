<div class="mw-components">
    <m-expandable-layout class="mw-components__body"
                         :open="showMenu"
                         panel-width="320px">
        <div class="mw-components__page">
            <m-icon-button class="mw-components__menu-button"
                           @click="toggleMenu()"
                           :icon-name="showMenu ? 'm-svg__close-clear' : 'm-standard-edit'"
                           skin="light"
                           icon-size="20px">
            </m-icon-button>

            <div class="mw-components__content">
                <mw-component-details :component="component"></mw-component-details>
            </div>
        </div>
        <template slot="panel">
            <div class="mw-components__sidemenu">

                <div class="mw-components__search">
                    <m-searchfield></m-searchfield>
                </div>
                <div class="mw-components__sidemenu-body">
                    <div class="mw-components__sidemenu-title">
                        <m-i18n k="website:components-ui"></m-i18n>
                    </div>
                    <m-accordion skin="plain"
                                 icon-position="right"
                                 icon-size="small"
                                 v-for="categoryId in componentCategoryIds"
                                 :key="categoryId"
                                 :open="urlMatchPath(categoryId)">
                        <div class="mw-components__sidemenu-sub-title"
                             slot="header">
                            <m-i18n :k="'categories:'+categoryId"></m-i18n>
                        </div>
                        <div>
                            <div class="mw-components__sidemenu-item"
                                 v-for="component in getCategoryComponents(categoryId)"
                                 :key="component.url">
                                <router-link :to="getRouterIndex(component.url)"
                                             @click.native="onSideMenuSelection()"><span>
                                        {{component.name}}
                                    </span>
                                    <span v-if="component.beta">
                                        <span class="modul-logo__beta mw-components__beta">
                                            <m-i18n k="website:beta"></m-i18n>
                                        </span>
                                    </span>
                                </router-link>
                            </div>
                        </div>
                    </m-accordion>
                </div>
            </div>
        </template>
    </m-expandable-layout>
</div>
