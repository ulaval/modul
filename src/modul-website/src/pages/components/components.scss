@import '../../styles/commons';

.mw-component-details {
    width: 100%;

    &__component-navigation {
        display: flex;
        margin-top: 54px;
        color: $m-color--interactive-light;
    }

    &__component-navigation-next,
    &__component-navigation-prev {
        line-height: 24px;
        position: relative;
        padding: 1rem;
        flex-grow: 1;
        border: 1px solid;
        text-align: center;
        text-decoration: none;
        font-weight: $m-font-weight--semi-bold;
    }

    &__component-navigation-next--icon {
        position: absolute;
        right: 16px;
    }

    &__component-navigation-prev--icon {
        position: absolute;
        left: 16px;
    }
}
