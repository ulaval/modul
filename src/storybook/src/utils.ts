export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';
export const modulWebsiteHierarchyRootSeparator: string = 'modul-website|';
