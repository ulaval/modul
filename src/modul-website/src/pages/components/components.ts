
import { MWExpandablePanel } from '@/components/expendable-panel/expandable-panel';
import { ModulWebsite } from '@/components/modul-website';
import { ComponentMeta } from '@/content/components.meta.loader';
import _ from 'lodash';
import Vue from 'vue';
import Component from 'vue-class-component';
import { Prop, Watch } from 'vue-property-decorator';
import { MWComponent } from './component/component-details';
import WithRender from './components.html?style=./components.scss';

@WithRender
@Component({
    components: { 'mw-component-details': MWComponent }
})
export class MWComponentsPage extends ModulWebsite {

    @Prop()
    component: ComponentMeta;

    $refs: {
        expandablePanel: MWExpandablePanel;
        prevLink: Vue;
        nextLink: Vue;
    };

    fullPath: string;

    get componentCategoryIds(): string[] {
        return _.sortBy(this.$meta.categories, name => _.deburr(this.$i18n.translate(`categories:${name}`)));
    }

    get componentsNavigationOrder(): ComponentMeta[] {
        let components: ComponentMeta[] = [];
        this.componentCategoryIds.forEach(category => {
            this.$meta.componentState[category].forEach((componentMeta: ComponentMeta) => {
                components.push(componentMeta);
            });
        });
        return components;
    }

    get currentComponentNavigationIndex(): number {
        return _.findIndex(this.componentsNavigationOrder, (c: ComponentMeta) => {
            return this.component.name === c.name;
        });
    }

    get prevComponent(): ComponentMeta {
        let index = this.currentComponentNavigationIndex;

        // if this is the first item return the last
        if (index - 1 < 0) {
            return this.componentsNavigationOrder[this.componentsNavigationOrder.length - 1];
        } else {
            return this.componentsNavigationOrder[index - 1];
        }
    }

    get nextComponent(): ComponentMeta {
        let index = this.currentComponentNavigationIndex;

        // if this is the last item return the first
        if (index + 1 >= this.componentsNavigationOrder.length - 1) {
            return this.componentsNavigationOrder[0];
        } else {
            return this.componentsNavigationOrder[index + 1];
        }
    }

    getRouterIndex(tag: string): string {
        return this.$routerIndex.for(tag);
    }

    getCategoryComponents(category): ComponentMeta[] {
        return this.$meta.componentState[category];
    }

    urlMatchPath(categoryId: string) {
        let path: string = this.$i18n.translate(`categories:${categoryId}-route`);
        return this.fullPath.includes(`/${path}`);
    }

    onSideMenuSelection() {
        this.$refs.expandablePanel.onSideMenuSelection();
    }

    onPrevLinkClick() {
        (this.$refs.prevLink.$el as HTMLElement).blur();
    }

    onNextLinkClick() {
        (this.$refs.nextLink.$el as HTMLElement).blur();
    }

    @Watch('$route', { immediate: true })
    private routeChanged(): void {
        this.fullPath = this.$route.fullPath;
    }
}
