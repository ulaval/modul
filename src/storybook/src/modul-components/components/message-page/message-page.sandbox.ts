// tslint:disable-next-line: import-blacklist
import { MButton } from '@ulaval/modul-components/dist/components/button/button';
import { BUTTON_NAME, MESSAGE_PAGE_NAME } from '@ulaval/modul-components/dist/components/component-names';
import MessagePagePlugin, { Link } from '@ulaval/modul-components/dist/components/message-page/message-page';
import { MMessageState } from '@ulaval/modul-components/dist/components/message/message';
import PopupDirectivePlugin from '@ulaval/modul-components/dist/directives/popup/popup';
// tslint:disable-next-line: import-blacklist
import moment from 'moment';
import Vue, { PluginObject } from 'vue';
import { Component } from 'vue-property-decorator';
import WithRender from './message-page.sandbox.html';

@WithRender
@Component({
    components: {
        [BUTTON_NAME]: MButton
    }
})
export class MMessagePageSandbox extends Vue {
    errorType: MMessageState = MMessageState.Error;
    warningType: MMessageState = MMessageState.Warning;
    informationType: MMessageState = MMessageState.Information;
    title: string = 'My error title';
    oneHint: string[] = ['My only hint'];
    manyHints: string[] = ['My first hint', 'My second long hint.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla egestas urna rhoncus ipsum congue lobortis. Mauris vel neque condimentum, dignissim lectus ac, vehicula justo. Aliquam nunc leo, tristique hendrerit aliquam sagittis, scelerisque quis libero. Integer non augue nec lacus aliquet porttitor in nec lacus. Cras ultrices tellus est, condimentum gravida orci pulvinar quis. Integer eget turpis arcu. Curabitur consequat porta urna, at hendrerit justo consectetur non. Aenean venenatis ornare nulla, a vulputate erat eleifend ut.'];
    oneLink: Link[] = [new Link('the only link', 'http://www.ulaval.ca')];
    manyLinks: Link[] = [new Link('the first link', 'http://www.ulaval.ca', true), new Link('the second link', 'http://www.google.com', true)];
    errorDate: moment.Moment = moment().subtract(moment.duration(2, 'days'));
    stackTrace: string = `(l1) This is a multiline stack
(l2) to be displayed
(l3) in the error page.`;
}

const MessagePageSandboxPlugin: PluginObject<any> = {
    install(v, options): void {
        v.use(PopupDirectivePlugin);
        v.use(MessagePagePlugin);
        v.component(`${MESSAGE_PAGE_NAME}-sandbox`, MMessagePageSandbox);
    }
};

export default MessagePageSandboxPlugin;
