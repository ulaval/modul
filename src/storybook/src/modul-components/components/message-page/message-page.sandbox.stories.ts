import { MESSAGE_PAGE_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { sandboxDecorator } from '../../../sandbox-decorator';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${MESSAGE_PAGE_NAME}`
};

export const sandbox: any = () => '<m-message-page-sandbox></m-message-page-sandbox>';

sandbox.story = {
    decorators: [sandboxDecorator],
    parameters: { fileName: __filename, options: { showPanel: false, isToolshown: true } }
};
