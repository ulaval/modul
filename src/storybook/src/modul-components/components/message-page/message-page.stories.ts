import { array, select, text } from '@storybook/addon-knobs';
import { MESSAGE_PAGE_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { Link, MMessagePageSkin } from '@ulaval/modul-components/dist/components/message-page/message-page';
import { MMessageState } from '@ulaval/modul-components/dist/components/message/message';
import { Enums } from '@ulaval/modul-components/dist/utils/enums/enums';
import { ModulIconName } from '@ulaval/modul-components/dist/utils/modul-icons/modul-icons';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${MESSAGE_PAGE_NAME}`,
    parameters: { fileName: __filename }
};

export const defaultStory = () => ({
    props: {
        iconName: {
            default: text(
                'Prop icon-name',
                ''
            )
        },
        state: {
            default: select<MMessageState>(
                'Prop state',
                Enums.toValueArray(MMessageState) as MMessageState[],
                MMessageState.Error
            )
        },
        title: {
            default: text(
                'Prop title',
                ''
            )
        },
        skin: {
            default: select<MMessagePageSkin>(
                'Prop skin',
                Enums.toValueArray(MMessagePageSkin) as MMessagePageSkin[],
                MMessagePageSkin.Default
            )
        },
        hints: {
            default: array(
                'Prop hints',
                [
                    'Lorem ipsum dolor sit amet.', 'Excepteur sint occaecat cupidatat non.'
                ]
            )
        }
    },
    template: `<${MESSAGE_PAGE_NAME}
        :icon-name="iconName"
        :state="state"
        :skin="skin"
        :title="title"
        :hints="hints"
    />`
});

defaultStory.story = {
    name: 'Sandbox'
};

export const Skin = () => ({
    data: () => ({
        skins: Enums.toValueArray(MMessagePageSkin) as MMessagePageSkin[]
    }),
    template: `<div>
        <${MESSAGE_PAGE_NAME}
            v-for="(skin, index) in skins"
            :key="index"
            :class="{ 'm-u--margin-top--l': index > 0 }"
            :skin="skin"
            :title="'Prop skin = ' + skin"
        />
    </div>`
});


export const State = () => ({
    data: () => ({
        states: Enums.toValueArray(MMessageState) as MMessageState[]
    }),
    template: `<div>
        <${MESSAGE_PAGE_NAME}
            v-for="(state, index) in states"
            :key="index"
            :class="{ 'm-u--margin-top--xl': index > 0 }"
            :state="state"
            :title="'Prop state = ' + state"
        />
    </div>`
});

export const IconName = () => ({
    props: {
        iconName: {
            default: text(
                'Prop icon-name',
                ModulIconName.Edit
            )
        }
    },
    template: `<${MESSAGE_PAGE_NAME}
        :icon-name="iconName"
        state="${MMessageState.Confirmation}"
    />`
});


export const Hints = () => ({
    props: {
        hints: {
            default: array(
                'Prop hints',
                [
                    'Lorem ipsum dolor sit amet.', 'Excepteur sint occaecat cupidatat non.'
                ]
            )
        }
    },
    template: `<${MESSAGE_PAGE_NAME}
        :hints="hints"
        state="${MMessageState.Confirmation}"
    />`
});


export const Links = () => ({
    data: () => ({
        links: [new Link('Google', 'www.google.ca', true), new Link('FaceBook', 'www.facebook.com', true)]
    }),
    template: `<${MESSAGE_PAGE_NAME}
        :links="links"
        state="${MMessageState.Confirmation}"
    />`
});

export const DefaultSlot = () => ({
    template: `<${MESSAGE_PAGE_NAME}
        state="${MMessageState.Information}"
    >
        Default slot
    </${MESSAGE_PAGE_NAME}>`
});
