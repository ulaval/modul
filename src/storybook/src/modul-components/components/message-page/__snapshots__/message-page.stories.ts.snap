// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-message-page Default Slot 1`] = `
<article
  class="m-message-page m--is-state-information m--is-skin-default"
>
  <header
    class="m-message-page__header"
  >
    <!---->
     
    <div
      aria-hidden="true"
      class="m-message-page__icon-container"
    >
      <svg
        aria-hidden="true"
        class="m-icon m-message-page__icon"
        height="130px"
        width="130px"
      >
        <!---->
         
        <use
          aria-hidden="true"
          class=""
        />
      </svg>
    </div>
  </header>
   
  <div
    class="m-message-page__body"
  >
     
    <div
      class="m-message-page__details"
    >
      
        Default slot
    
    </div>
     
    <!---->
  </div>
</article>
`;

exports[`Storyshots modul-components|m-message-page Hints 1`] = `
<article
  class="m-message-page m--is-state-confirmation m--is-skin-default"
>
  <header
    class="m-message-page__header"
  >
    <!---->
     
    <div
      aria-hidden="true"
      class="m-message-page__icon-container"
    >
      <svg
        aria-hidden="true"
        class="m-icon m-message-page__icon"
        height="130px"
        width="130px"
      >
        <!---->
         
        <use
          aria-hidden="true"
          class=""
        />
      </svg>
    </div>
  </header>
   
  <div
    class="m-message-page__body"
  >
    <p
      class="m-message-page__hints m--no-content"
    >
      Lorem ipsum dolor sit amet.
    </p>
    <p
      class="m-message-page__hints m--no-content"
    >
      Excepteur sint occaecat cupidatat non.
    </p>
     
    <!---->
     
    <!---->
  </div>
</article>
`;

exports[`Storyshots modul-components|m-message-page Icon Name 1`] = `
<article
  class="m-message-page m--is-state-confirmation m--is-skin-default"
>
  <header
    class="m-message-page__header"
  >
    <!---->
     
    <div
      aria-hidden="true"
      class="m-message-page__icon-container"
    >
      <svg
        aria-hidden="true"
        class="m-icon m-message-page__icon"
        height="130px"
        width="130px"
      >
        <!---->
         
        <use
          aria-hidden="true"
          class=""
        />
      </svg>
    </div>
  </header>
   
  <!---->
</article>
`;

exports[`Storyshots modul-components|m-message-page Links 1`] = `
<article
  class="m-message-page m--is-state-confirmation m--is-skin-default"
>
  <header
    class="m-message-page__header"
  >
    <!---->
     
    <div
      aria-hidden="true"
      class="m-message-page__icon-container"
    >
      <svg
        aria-hidden="true"
        class="m-icon m-message-page__icon"
        height="130px"
        width="130px"
      >
        <!---->
         
        <use
          aria-hidden="true"
          class=""
        />
      </svg>
    </div>
  </header>
   
  <div
    class="m-message-page__body"
  >
     
    <!---->
     
    <ul
      class="m-message-page__links"
    >
      <li>
        <a
          class="m-link m-message-page__link m--has-left-icon m--has-content"
          href="www.google.ca"
          tabindex="0"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="m-icon m-link__icon m--has-large-stroke"
            height="12px"
            width="12px"
          >
            <!---->
             
            <use
              aria-hidden="true"
              class=""
            />
          </svg>
           
          <span
            class="m-link__text"
          >
            Google
          </span>
           
          <span
            class="m-i18n m-link__hidden"
          >
            Cet hyperlien s'ouvrira dans un nouvel onglet.
          </span>
        </a>
      </li>
      <li>
        <a
          class="m-link m-message-page__link m--has-left-icon m--has-content"
          href="www.facebook.com"
          tabindex="0"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="m-icon m-link__icon m--has-large-stroke"
            height="12px"
            width="12px"
          >
            <!---->
             
            <use
              aria-hidden="true"
              class=""
            />
          </svg>
           
          <span
            class="m-link__text"
          >
            FaceBook
          </span>
           
          <span
            class="m-i18n m-link__hidden"
          >
            Cet hyperlien s'ouvrira dans un nouvel onglet.
          </span>
        </a>
      </li>
    </ul>
  </div>
</article>
`;

exports[`Storyshots modul-components|m-message-page Sandbox 1`] = `
<article
  class="m-message-page m--is-state-error m--is-skin-default"
>
  <header
    class="m-message-page__header"
  >
    <!---->
     
    <div
      aria-hidden="true"
      class="m-message-page__icon-container"
    >
      <svg
        aria-hidden="true"
        class="m-icon m-message-page__icon"
        height="130px"
        width="130px"
      >
        <!---->
         
        <use
          aria-hidden="true"
          class=""
        />
      </svg>
    </div>
  </header>
   
  <div
    class="m-message-page__body"
  >
    <p
      class="m-message-page__hints m--no-content"
    >
      Lorem ipsum dolor sit amet.
    </p>
    <p
      class="m-message-page__hints m--no-content"
    >
      Excepteur sint occaecat cupidatat non.
    </p>
     
    <!---->
     
    <!---->
  </div>
</article>
`;

exports[`Storyshots modul-components|m-message-page Skin 1`] = `
<div>
  <article
    class="m-message-page m--is-state-error m--is-skin-default"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        Prop skin = default
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <svg
          aria-hidden="true"
          class="m-icon m-message-page__icon"
          height="130px"
          width="130px"
        >
          <!---->
           
          <use
            aria-hidden="true"
            class=""
          />
        </svg>
      </div>
    </header>
     
    <!---->
  </article>
  <article
    class="m-message-page m--is-state-error m--is-skin-light m-u--margin-top--l"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        Prop skin = light
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <svg
          aria-hidden="true"
          class="m-icon m-message-page__icon"
          height="76px"
          width="76px"
        >
          <!---->
           
          <use
            aria-hidden="true"
            class=""
          />
        </svg>
      </div>
    </header>
     
    <!---->
  </article>
</div>
`;

exports[`Storyshots modul-components|m-message-page State 1`] = `
<div>
  <article
    class="m-message-page m--is-state-confirmation m--is-skin-default"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        Prop state = confirmation
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <svg
          aria-hidden="true"
          class="m-icon m-message-page__icon"
          height="130px"
          width="130px"
        >
          <!---->
           
          <use
            aria-hidden="true"
            class=""
          />
        </svg>
      </div>
    </header>
     
    <!---->
  </article>
  <article
    class="m-message-page m--is-state-information m--is-skin-default m-u--margin-top--xl"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        Prop state = information
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <svg
          aria-hidden="true"
          class="m-icon m-message-page__icon"
          height="130px"
          width="130px"
        >
          <!---->
           
          <use
            aria-hidden="true"
            class=""
          />
        </svg>
      </div>
    </header>
     
    <!---->
  </article>
  <article
    class="m-message-page m--is-state-warning m--is-skin-default m-u--margin-top--xl"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        Prop state = warning
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <svg
          aria-hidden="true"
          class="m-icon m-message-page__icon"
          height="130px"
          width="130px"
        >
          <!---->
           
          <use
            aria-hidden="true"
            class=""
          />
        </svg>
      </div>
    </header>
     
    <!---->
  </article>
  <article
    class="m-message-page m--is-state-error m--is-skin-default m-u--margin-top--xl"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        Prop state = error
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <svg
          aria-hidden="true"
          class="m-icon m-message-page__icon"
          height="130px"
          width="130px"
        >
          <!---->
           
          <use
            aria-hidden="true"
            class=""
          />
        </svg>
      </div>
    </header>
     
    <!---->
  </article>
</div>
`;
